test/realm_test.dart

```diff
@@ -595,4 +598,43 @@ Future<void> main([List<String>? args]) async {
     expect(mainSchools[0].branches[0].students.length + mainSchools[0].branches[1].students.length, 3);
     realm.close();
   });
+
+  test('Realm.operator==', () {
+    var config = Configuration([Dog.schema, Person.schema]);
+    final realm1 = Realm(config);
+    config = Configuration([Dog.schema, Person.schema]);
+    final realm2 = Realm(config);
+
+    expect(realm1, realm1);
+    expect(realm2, realm2);
+    expect(realm1, isNot(realm2));
+
+    realm1.write(() {
+      realm1.add(Person('Kasper'));
+    });
+
+    final realm3 = realm1.all<Person>().first.realm;
```

Comment:
This test does too much. This is testing RealmResults. I suggest we remove this.

Reply:
OK